scripts/policy/protocols/ssl/decryption.zeek

```diff
@@ -0,0 +1,112 @@
+##! This script allows for the decryption of certain TLS 1.2 connection, if the user is in possession
+##! of the private key material for the session. Key material can either be provided via a file (useful
+##! for processing trace files) or via sending events via broker (for live decoding).
+##!
+##! Please note that this feature is experimental and can change without guarantees to our typical
+##! deprecation tieline. Please also note that currently only TLS 1.2 connections that use the
+##! TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 cipher suite are supported.
+
+@load base/frameworks/input
+@load base/frameworks/notice
+@load base/protocols/conn
+@load base/protocols/ssl
+
+module SSL;
+
+# Do not disable analyzers after detection - otherwise we will not receive
+# encrypted packets.
+redef SSL::disable_analyzer_after_detection = F;
+
+export {
+	## This can be set to a file that contains the session secrets for decryption, when parsing a pcap file.
+	## Please note that, when using this feature, you probably want to pause processing of data till this
+	## file has been read.
+	const keylog_file = getenv("ZEEK_TLS_KEYLOG_FILE") &redef;
+
+	## Secrets expire after this time of not being used.
+	const secret_expiration = 5 mins &redef;
+
+	## This event can be triggered, e.g., via broker to add known keys to the TLS key database.
+	##
+	## client_random: client random for which the key is set
+	##
+	## keys: key material
+	global add_keys: event(client_random: string, keys: string);
+
+	## This event can be triggered, e.g., via broker to add known secrets to the TLS secret datbase.
+	##
+	## client_random: client random for which the secret is set
+	##
+	## secrets: derived TLS secrets material
+	global add_secret: event(client_random: string, secret: string);
+}
+
+@if ( keylog_file == "" )
+# If a keylog file was given via an environment variable, let's disable secret expiration - that does not
+# make sense for pcaps.
+global secrets: table[string] of string = {} &redef;
+global keys: table[string] of string = {} &redef;
+@else
+#global secrets: table[string] of string = {} &read_expire=secret_expiration &redef;
+#global keys: table[string] of string = {} &read_expire=secret_expiration &redef;
```

Comment:
Is this supposed to be commented out?

Reply:
no, it is not - thank you. I clearly should (and will) add another test.